core/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -1089,6 +1094,16 @@ private static String arrayToParsableString(String[] array) {
         return timeSetting(key, defaultValue, TimeValue.timeValueMillis(0), properties);
     }
 
+    public static Setting<TimeValue> timeSettingWithNegativeValuesDeprecated(String key, TimeValue defaultValue, Property... properties) {
```

Comment:
I prefer not adding abstractions with exactly one use case. Especially since when we turn around and remove the deprecation logging for this setting (and turn it into a hard failure with `Setting#positiveTimeSetting`), the abstraction might end up left behind.

This one is also dangerous because it adds a static logger to a class that is touched very early. If we are not careful, we start initializing logging before we have configured it which we actively try to detect and fail startup if it occurs.

I think that you can get by with manually defining the logic for this setting at the definition site.